File: backends/spring-kotlin/src/main/kotlin/dev/stackverse/backend/message/MessageDtos.kt
package dev.stackverse.backend.message

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.Instant
import java.util.UUID

data class MessageRequest(
    val key: String? = null,
    val language: String? = null,
    val text: String? = null,
    val description: String? = null,
)

@JsonInclude(JsonInclude.Include.NON_NULL)
data class MessageResponse(
    val id: UUID,
    val key: String,
    val language: String,
    val text: String,
    val description: String?,
    val createdAt: Instant,
    val updatedAt: Instant,
) {
    companion object {
        fun of(message: Message) = MessageResponse(
            id = message.id,
            key = message.key,
            language = message.language,
            text = message.text,
            description = message.description,
            createdAt = message.createdAt,
            updatedAt = message.updatedAt,
        )
    }
}

data class MessageBundleResponse(
    val language: String,
    val messages: Map<String, String>,
)

File: backends/dotnet/src/StackverseBackend/Messages/MessageDtos.cs
namespace StackverseBackend.Messages;

public sealed record MessageRequest(
    string? Key = null,
    string? Language = null,
    string? Text = null,
    string? Description = null);

public sealed record MessageResponse(
    Guid Id,
    string Key,
    string Language,
    string Text,
    string? Description,
    DateTime CreatedAt,
    DateTime UpdatedAt)
{
    public static MessageResponse Of(Message message) => new(
        message.Id,
        message.Key,
        message.Language,
        message.Text,
        message.Description,
        message.CreatedAt,
        message.UpdatedAt);
}

public sealed record MessageBundleResponse(string Language, IReadOnlyDictionary<string, string> Messages);
